lib/fieldTypes/azurefile.js

```diff
@@ -0,0 +1,358 @@
+/*!
+ * Module dependencies.
+ */
+
+var _ = require('underscore'),
+	moment = require('moment'),
+	keystone = require('../../'),
+	async = require('async'),
+	util = require('util'),
+	azure = require('azure'),
+	utils = require('keystone-utils'),
+	super_ = require('../field');
+
+
+/**
+ * AzureFile FieldType Constructor
+ * @extends Field
+ * @api public
+ */
+
+function azurefile(list, path, options) {
+	var self = this;
+	this._underscoreMethods = ['format', 'uploadFile'];
+	
+	// event queues
+	this._pre = {
+		upload: []
+	};
+	
+	// TODO: implement filtering, usage disabled for now
+	options.nofilter = true;
+
+	// TODO: implement initial form, usage disabled for now
+	if (options.initial) {
+		throw new Error('Invalid Configuration\n\nAzureFile fields (' + list.key + '.' + path + ') do not currently support being used as initial fields.\n');
+	}
+	
+	azurefile.super_.call(this, list, path, options);
+	
+	// validate azurefile config (has to happen after super_.call)
+	if (!this.azurefileconfig) {
+		throw new Error('Invalid Configuration\n\n' +
+			'AzureFile fields (' + list.key + '.' + path + ') require the "azurefile config" option to be set.\n\n' + 
+			'See http://keystonejs.com/docs/configuration#azurefile for more information.\n');
+	}
+
+	process.env.AZURE_STORAGE_ACCOUNT = this.azurefileconfig.account;
+	process.env.AZURE_STORAGE_ACCESS_KEY = this.azurefileconfig.key;
+
+	self.azurefileconfig.container = self.azurefileconfig.container || 'keystone';
+
+	options.filenameFormatter = options.filenameFormatter || function(item, filename) { return filename };
+	options.containerFormatter = options.containerFormatter || function(item, filename) { return self.azurefileconfig.container };
```

Comment:
options.filenameFormatter and options.containerFormatter are never assigned to self.
Am i missing something ?